File: Algoritmo e Estrutura de Dados 2/Trabalho Prático 1/Palindromo.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

bool isPalind (char palavra[]){
    int tam = strlen(palavra) - 1,
        i = 0;

    bool isPalind = true;

    do{
        if (palavra [i] != palavra [tam]){
            isPalind = false;}
        i++;
        tam--;
    }while (i < tam && isPalind == true);
    return isPalind;
}

int main(){

    char frase[1000];

    scanf("%[^\n]%*c", &frase);
    
    while(strcmp(frase, "FIM") != 0){
        
        if(isPalind(frase) == true)
            printf("SIM \n");
            else printf("NAO \n");
            
            scanf("%[^\n]%*c",&frase);
    }   
    return 0;
}

File: Algoritmo e Estrutura de Dados 2/Trabalho Prático 1/Palindromo.java
class Palindromo {
    public static boolean isPalind (String s){
        int tam = s.length() - 1,
        i = 0;

        boolean ehPalind = true;

        do {
            if (s.charAt(i) != s.charAt(tam)){
                ehPalind = false;}
            i++;
            tam--;
        }while (i < tam && ehPalind==true);

        return ehPalind;
    }
  
    public static boolean isFim(String s){
       return (s.length() >= 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    public static void Saida (boolean isPalind)
    {
        if (isPalind==true)
        {
            MyIO.println("SIM");
        }
        else MyIO.println("NAO");
    }
  
    public static void main (String[] args){
       String[] entrada = new String[1000];
       int numEntrada = 0;
  
       //Leitura da entrada padrao
       do {
          entrada[numEntrada] = MyIO.readLine();
       } while (isFim(entrada[numEntrada++]) == false);
       numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM
  
       //Para cada linha de entrada, gerando uma de saida contendo o numero de letras maiusculas da entrada
       for(int i = 0; i < numEntrada; i++){
            Saida(isPalind(entrada[i]));  
       }
    }
 }